numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -2510,13 +2510,76 @@ gentype_getcharbuf(PyObject *self, Py_ssize_t segment, constchar **ptrptr)
 static int
 gentype_getbuffer(PyObject *self, Py_buffer *view, int flags)
 {
-    Py_ssize_t len;
-    void *buf;
+    PyArray_Descr *outcode;
+
+    if (flags & PyBUF_WRITABLE) {
+        PyErr_SetString(PyExc_BufferError, "scalar buffer is readonly");
+        view->obj = NULL;
+        return -1;
+    }
+
+    outcode = PyArray_DescrFromScalar(self);
+
+    if (flags & PyBUF_FORMAT && 3 <= sizeof view->internal ) {
```

Comment:
Parens around `(a & b)` would be nice.

Are you sure `sizeof(view.internal)` does anything useful here?

Reply:
Currently pointless, but the compiler will optimize that away.

I was trying to guard against the scenario were code strings are 5 bytes long, and then it works on the 64bit build but not the 32bit build. Having an explicit call to `sizeof` was meant to be a reminder to the programmer not to go past their limit. That might be unnecessary/ineffective defensive programming.

It would probably be better to do `memset(&view->internal, '\0', sizeof view->internal)`. That might get the same idea across and zero out the everything instead of the first 3 bytes.